crates/optimism/rpc/src/eth/mod.rs

```diff
@@ -273,6 +278,85 @@ impl<N: OpNodeCore> fmt::Debug for OpEthApi<N> {
     }
 }
 
+#[async_trait::async_trait]
+impl<N> L2EthApiExtServer for OpEthApi<N>
+where
+    N: OpNodeCore + 'static,
+    N::Provider: BlockReaderIdExt + StateProviderFactory,
+{
+    async fn send_raw_transaction_conditional<'life0>(
+        &'life0 self,
+        bytes: Bytes,
+        condition: TransactionConditional,
+    ) -> RpcResult<B256> {
+        // calculate and validate cost
+        let cost = condition.cost();
+        if cost > MAX_CONDITIONAL_EXECUTION_COST {
+            return Err(OpEthApiError::InvalidTransaction(
+                OpInvalidTransactionError::ConditionalCostExceeded,
+            )
+            .into());
+        }
+
+        // get current header
+        let header = self.provider().latest_header().map_err(OpEthApiError::Eth)?.header();
+
+        // check condition against header
+        if !condition.matches_block_number(header.number) ||
+            !condition.matches_timestamp(header.timestamp)
```

Comment:
there's a subtle nuance here because `matche` is intended for inclusion check, I think this could be problematic if for example only the min block value is set?

there are `has_exceeded_*` calls that check whether the conditional con no longer be included